src/design_patterns/Shipper.ts

```diff
@@ -0,0 +1,113 @@
+import ShipentKinds, {IShipmentKinds, Letter, Package, Oversize} from './ShipmentKinds'
+
+
+interface IShipper {
+    getCost(): number
+}
+
+enum Reates {
+    AIR_EAST = 0.39,
+    CHICAGO = 0.42, 
+    PACIFIC_PARCEL = 0.51
+}
+
+type ShipperItemKinds = typeof Letter | typeof Package | typeof Oversize
+
+const isOversize = (itemKind: ShipperItemKinds): itemKind is typeof Oversize => {
+    return !!(itemKind as typeof Oversize).oversizeAirEast
+}
+
+export class AirEastShipper implements IShipper {
+
+    private itemWeight: number
+    private readonly rate: number = Reates.AIR_EAST
+    private kind: ShipperItemKinds
+
+    constructor(wight: number, kind: ShipperItemKinds){
```

Comment:
IMO if `kind` it's a class, it's better to name it with Capital letter.

Reply:
fixed